components/tidb_query/src/rpn_expr/impl_string.rs

```diff
@@ -239,6 +239,38 @@ pub fn locate_binary_3_args(
     }
 }
 
+#[rpn_fn(varg, min_args = 1)]
+#[inline]
+pub fn field_int(args: &[&Option<Int>]) -> Result<Option<Int>> {
```

Comment:
I think you can write

```rust
#[rpn_fn(varg, min_args = 1)]
#[inline]
pub fn field<T: Evaluable + PartialEq>(args: &[&Option<T>]) -> Result<Option<Int>> { .. }
```

directly. You can assign it in the mapping using `ScalarFuncSig::FieldInt => field_fn_meta::<Int>()`

Reply:
ok